### PR TITLE
All non-base layers are data layers

### LINKED ISSUE #10332: Review update_table_dependencies for Layer model
In the old Layer model, the table dependencies are only updated for data_layers: 
https://github.com/CartoDB/cartodb/blob/4ab2e145e728f7c03be14289bb1b910e9fcd70a4/app/models/layer.rb#L87

In Carto::Layer, `data_layer?` means carto or torque, but on `::Layer`, it means carto. So this is most likely incorrect. We probably need to fix it, so table dependencies are updated for torque layers as well.

cc @CartoDB/builder-backend

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #10332 and #10368.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@javitonino CR this, please.

File: app/models/map/copier.rb [hunk 1]
```diff
@@ -33,9 +33,32 @@ def copy_base_layer(origin_map, destination_map)
...
+                  first_base = destination_map.layers.find(&:base_layer?)
+                  first_base ? (first_base.order + 1) : 0
+                end
+
         data_layer_copies_from(origin_map, user).map do |layer|
+          # Push layers on top if needed
+          if(destination_map.layers.map(&:order).include?(order))
+            destination_map.layers.select { |l| l.order >= order }.each do |layer|
+              layer.order += 1
+              layer.save
```
Comment (javitonino) [inline app/models/map/copier.rb]: If creating a map from several layers, this layer will be saved several times (and that triggers invalidations). Can we defer that save to after the loop?

**Comment by juanignaciosl** [comment]
@javitonino next CR, please.

**Comment by juanignaciosl** [comment]
@javitonino I had to bring older saves back because some tests failed. 👍 ?

**Comment by javitonino** [comment]
No problem :+1:

**Pull request merged**